### ISSUE TITLE
Alamofire doesn't properly report download progress on some URLs

### ISSUE THREAD

**Issue by michalrentka** [Alamofire/Alamofire#3526]
* [x] I've read, understood, and done my best to follow the [*CONTRIBUTING guidelines](https://github.com/Alamofire/Alamofire/blob/master/CONTRIBUTING.md).

## What did you do?

Download a remote file while reporting progress.

## What did you expect to happen?

Proper increasing value of `Progress.fractionCompleted`

## What happened instead?

Alamofire keeps reporting `Progress.fractionCompleted = 0`. `Progress.totalUnitCount = -1` even though `Progress.completedUnitCount` is increasing properly.

## Alamofire Environment

Alamofire Version: 5.4.4
Dependency Manager: SwiftPM
Xcode Version: 13.1
Swift Version: 5.5.1
Platform(s) Running Alamofire: iOS
macOS Version Running Xcode: 12.0.1

## Demo Project

To reproduce the issue you can simply try downloading this url: `https://www.science.org/doi/pdf/10.1126/sciadv.abl8213`

```
let downloadRequest = AF.download(URLRequest(url: URL(string: "https://www.science.org/doi/pdf/10.1126/sciadv.abl8213")!))
downloadRequest.downloadProgress { progress in
    NSLog("AF PROGRESS: \(progress.fractionCompleted); \(progress.totalUnitCount)")
}
downloadRequest.responseData { _ in
}
downloadRequest.resume()
```
Above code reports `AF PROGRESS: 0.0; -1`.

`URLSession` handles this correctly:
```
let task = URLSession.shared.downloadTask(with: URL(string: "https://www.science.org/doi/pdf/10.1126/sciadv.abl8213")!) { _, _, _ in
}
task.progress.observable.subscribe(onNext: { progress in
    NSLog("URLSESSION PROGRESS: \(progress.fractionCompleted)")
})
.disposed(by: self.disposeBag)
task.resume()
```
The `progress.observable` just observes `Progress.fractionCompleted` value.

**Comment by jshier** [comment]
This is expected. `science.org` doesn't return a `Content-Length` header and so `URLSessionDownloadTask` returns `-1` for the `totalUnitCount`, which means that the `fractionCompleted` is always 0. You can check the `completedUnitCount` to see how much data has actually been downloaded, as that value properly accumulates during the download.

```
9726/-1
9726/-1
11086/-1
13824/-1
16562/-1
```

I'm not sure what `progress.observable` does (can you link the source?) but printing the `URLSessionDownloadTask`'s `progress values directly actually gives a compound value.

```
<NSProgress: 0x102220a10> : Parent: 0x0 (portion: 0) / Fraction completed: 0.9500 / Completed: 95 of 100  
<NSProgress: 0x102220ab0> : Parent: 0x102220a10 (portion: 5) / Fraction completed: 0.0000 / Completed: 0 of 100
```
These values are repeated for the entire download, which is rather useless. It then completes with a final, single value.

```
<NSProgress: 0x102220a10> : Parent: 0x0 (portion: 0) / Fraction completed: 0.0000 / Completed: 195 of 100
```
Obviously it knows something has completed, but the progress is all wrong and so the `fractionCompleted` goes back to 0. 

In short, you need to check more than just the `fractionCompleted` to work with these `Progress` values correctly.
